merkledag/ipld.md

```diff
@@ -308,6 +308,116 @@ In order to preserve merkle-linking's power, we must ensure that there is a sing
 
 **The IPLD Canonical format is _canonicalized CBOR_.**
 
+## Relationship with Protocol Buffers legacy IPFS node format
+
+IPLD has a known conversion with the legacy Protocol Buffers format. This format is defined with the Protocol Buffers syntax as:
+
+    message PBLink {
+        optional bytes  Hash = 1;
+        optional string Name = 2;
+        optional uint64 Tsize = 3;
+    }
+    
+    message PBNode {
+        repeated PBLink Links = 2;
+        optional bytes  Data = 1;
+    }
+
+The conversion to the IPLD data model must have the following properties:
+
+- It should be convertible back to protocol buffers, resulting in an identical byte stream (so the hash corresponds). This implies that ordering and duplicate links must be preserved in some way.
+- When using paths as defined earlier in this document, links should be accessible without further indirection. This requires the top node object to have keys corresponding to link names.
+- Link names should not conflict with other keys.
+
+There are multiple ways to do that that will be described next.
+
+### Current encoding in go-ipld
+
+go-ipld implements the following conversion:
+
+    {
+      "<Links[0].Name.(escaped)>": {
+        "hash": "<Links[0].Hash>",
+        "name": "<Links[0].Name>",
+        "size": <Links[0].Tsize>
```

Comment:
That's right, but they currently aren't. This is something that needs to be modified in go-ipld (and in this spec as well).